stack-ghc-lib.yaml

```diff
@@ -14,8 +14,11 @@ extra-deps:
 - shake-0.18.5
 - regex-base-0.94.0.0
 - regex-tdfa-1.3.1.0
-- haddock-library-1.8.0
-- haddock-api-2.23.1
+- haddock-library-1.7.0
+- github: haskell/haddock
+  commit: 1f704e30d5f40fa5113045b252c0e48aebd842be
+  subdirs:
+    - haddock-api
```

Comment:
Why and how long do we need this custom version of haddock-api? I'm not keen on merging as it will prevent a Hackage release

Reply:
this was removed from stack-ghc-lib.yaml -  haddock-api uses ghc and can't be used from ghc-lib mode anyway (still uses old documentation lookup).

similar workaround with github dependency still there in stack-810.yaml:  haddock-api-2.24.0 from hackage doesn't compile with haddock-library-1.9.0 although it should.  
Same for 8.6 (stack.yaml)

This is easy to fix by making a new release of haddock-api upstream, but I'm not sure if that's the best way to go because it also will lock us with separate versions of `haddock-library` per GHC version